sync/rpc/server.go

```diff
@@ -282,7 +325,21 @@ func (*SyncAgentServer) BackupRemove(ctx context.Context, req *BackupRemoveReque
 	return &Empty{}, nil
 }
 
-func (*SyncAgentServer) BackupRestore(ctx context.Context, req *BackupRestoreRequest) (*Empty, error) {
+func (s *SyncAgentServer) BackupRestore(ctx context.Context, req *BackupRestoreRequest) (e *Empty, err error) {
+	if err := s.PrepareRestore(""); err != nil {
+		logrus.Errorf("failed to prepare backup restore: %v", err)
+		return nil, err
+	}
+
+	defer func() {
+		if err != nil {
+			// Reset the isRestoring flag to false
+			if err := s.FinishRestore(""); err != nil {
+				logrus.Errorf("failed to finish backup restore: %v", err)
+				return
+			}
+		}
+	}()
```

Comment:
Something seems wrong with this commit. Why it still contains `reexec.Command` below?

Reply:
This is because the order of commits has varied from my local git history to the one in pull request.
Is it okay if I squash the commits `Moving the RestoreBackup function to sync-agent-server` and `Moving the Incremental Restore function to Sync Agent` into `Sync-Agent server maintaining restore state` ?